packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+import { byteArrayEquals, toHexString } from '@chainsafe/ssz'
+import {
+  Address,
+  MAX_INTEGER,
+  bigIntToHex,
+  bigIntToUnpaddedBuffer,
+  bufferToBigInt,
+  bufferToHex,
+  computeVersionedHash,
+  ecrecover,
+  toBuffer,
+} from '@ethereumjs/util'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+
+import { BaseTransaction } from './baseTransaction'
+import { kzg } from './kzg/kzg'
+import {
+  BLOB_COMMITMENT_VERSION_KZG,
+  BlobNetworkTransactionWrapper,
+  BlobTransactionType,
+  LIMIT_BLOBS_PER_TX,
+  SignedBlobTransactionType,
+} from './types'
+import { AccessLists, checkMaxInitCodeSize } from './util'
+
+import type {
+  AccessList,
+  AccessListBuffer,
+  AccessListBufferItem,
+  BlobEIP4844TxData,
+  JsonTx,
+  TxOptions,
+  TxValuesArray,
+} from './types'
+import type { Common } from '@ethereumjs/common'
+
+const TRANSACTION_TYPE = 0x05
+const TRANSACTION_TYPE_BUFFER = Buffer.from(TRANSACTION_TYPE.toString(16).padStart(2, '0'), 'hex')
+
+const validateBlobTransactionNetworkWrapper = (
+  versionedHashes: Uint8Array[],
+  blobs: Uint8Array[],
+  commitments: Uint8Array[],
+  kzgProof: Uint8Array
+) => {
+  if (!(versionedHashes.length === blobs.length && blobs.length === commitments.length)) {
+    throw new Error('Number of versionedHashes, blobs, and commitments not all equal')
+  }
+
+  const verified = kzg.verifyAggregateKzgProof(blobs, commitments, kzgProof)
+
+  if (!verified) {
+    throw new Error('KZG proof cannot be verified from blobs/commitments')
+  }
+
+  for (let x = 0; x < versionedHashes.length; x++) {
+    const computedVersionHash = computeVersionedHash(commitments[x])
+    if (!byteArrayEquals(computedVersionHash, versionedHashes[x])) {
+      throw new Error(`commitment for blob at index ${x} does not match versionedHash`)
+    }
+  }
+}
+
+export class BlobEIP4844Transaction extends BaseTransaction<BlobEIP4844Transaction> {
```

Comment:
Class code doc is always relatively important for DevEx, since one is confronted with this pretty often, this should be added.